packages/muze-utils/src/common-utils.js

```diff
@@ -1467,6 +1467,56 @@ const nextAnimFrame = window.requestAnimationFrame || window.webkitRequestAnimat
         setTimeout(callback, 16);
     };
 
+/**
+ * Retrieves the nearest groupBy aggFn function in its derivation cycle for the specified field.
+ *
+ * @param {DataModel} dataModel - The target DataModel instance.
+ * @param {string} fieldName - The target field name.
+ * @return {string} Returns the aggFn name.
+ */
+const retrieveGroupByAggFn = (dataModel, fieldName) => {
+    let next = dataModel;
+    do {
+        const derivations = next.getDerivations();
+        if (derivations && derivations.length >= 1) {
+            const latestDerivation = derivations[derivations.length - 1];
+            if (latestDerivation.op === 'group') {
+                const aggFn = getObjProp(latestDerivation, 'criteria', fieldName);
+                if (aggFn) {
+                    return aggFn;
+                }
+            }
+        }
+    } while (next = next.getParent());
+
+    return null;
+};
+
+const retrieveNearestGroupByReducers = (dataModel) => {
```

Comment:
`retrieveNearestGroupByReducers` and `retrieveGroupByAggFn` almost have same code. Can't we make it a single method? If no `fieldName` is passed, then the method will return the aggregations functions of all measures

Reply:
yeah, then we can implement a method as `retrieveNearestGroupByReducers(datamodel, ...fieldNames)`